integration-tests/tests/circuits.rs

```diff
@@ -0,0 +1,85 @@
+macro_rules! declare_tests {
+    (($name:ident, $block_tag:expr),$mock:expr) => {
+        paste! {
+            #[tokio::test]
+            async fn [<serial_test_evm_ $name>]() {
+                log_init();
+                let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
+                test_evm_circuit_block(*block_num, $mock).await;
+            }
+
+            #[tokio::test]
+            async fn [<serial_test_state_ $name>]() {
+                log_init();
+                let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
+                test_state_circuit_block(*block_num, $mock).await;
+            }
+
+            #[tokio::test]
+            async fn [<serial_test_tx_ $name>]() {
+                log_init();
+                let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
+                test_tx_circuit_block(*block_num, $mock).await;
+            }
+
+            #[tokio::test]
+            async fn [<serial_test_bytecode_ $name>]() {
+                log_init();
+                let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
+                test_bytecode_circuit_block(*block_num, $mock).await;
+            }
+
+            #[tokio::test]
+            async fn [<serial_test_copy_ $name>]() {
+                log_init();
+                let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
+                test_copy_circuit_block(*block_num, $mock).await;
+            }
+        }
+    };
+}
+
+macro_rules! unroll_tests {
```

Comment:
@davidnevadoc  I don't understand the use of the word "unroll" here, is it something that means something in the context of Rust macros?

Reply:
Yes, the meaning of unroll here is related to Rust macros.  See this thread https://users.rust-lang.org/t/unroll-macro-argument-list/10510/2

Basically in the macro call you pass a list of values, and then the unroll generates some code for each value in the list.